framework/src/variables/MooseVariableDataLinearFV.C

```diff
@@ -0,0 +1,174 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "MooseVariableDataLinearFV.h"
+#include "MooseVariableDataFV.h"
+#include "MooseVariableField.h"
+#include "Assembly.h"
+#include "MooseError.h"
+#include "DisplacedSystem.h"
+#include "TimeIntegrator.h"
+#include "MooseVariableFV.h"
+#include "MooseTypes.h"
+#include "MooseMesh.h"
+#include "Attributes.h"
+#include "FVDirichletBCBase.h"
+#include "SubProblem.h"
+#include "FVKernel.h"
+#include "ADUtils.h"
+
+#include "libmesh/quadrature.h"
+#include "libmesh/fe_base.h"
+#include "libmesh/system.h"
+#include "libmesh/type_n_tensor.h"
+
+template <typename OutputType>
+MooseVariableDataLinearFV<OutputType>::MooseVariableDataLinearFV(
+    const MooseLinearVariableFV<OutputType> & var,
+    SystemBase & sys,
+    THREAD_ID tid,
+    Moose::ElementType element_type,
+    const Elem * const & elem)
+  : MooseVariableDataBase<OutputType>(var, sys, tid),
+    _var(var),
+    _fe_type(_var.feType()),
+    _var_num(_var.number()),
+    _assembly(_subproblem.assembly(_tid, var.kind() == Moose::VAR_SOLVER ? sys.number() : 0)),
+    _element_type(element_type),
+    _time_integrator(_sys.getTimeIntegrator()),
+    _elem(elem),
+    _displaced(dynamic_cast<const DisplacedSystem *>(&_sys) ? true : false),
+    _qrule(nullptr)
+{
+}
+
+template <typename OutputType>
+void
+MooseVariableDataLinearFV<OutputType>::setGeometry(Moose::GeometryType gm_type)
+{
+  switch (gm_type)
+  {
+    case Moose::Volume:
+    {
+      _qrule = _assembly.qRule();
+      // TODO: set integration multiplier to cell volume
```

Comment:
To be honest this part is copy-pasted from the NLFV part. I thought it would be wise to leave the comment there in case somebody wants to enhance the FE-like behavior of the variable this belongs to.